src/Bridges/Constraint/semi_to_binary.jl

```diff
@@ -0,0 +1,191 @@
+const SemiSets{T} = Union{MOI.Semicontinuous{T}, MOI.Semiinteger{T}}
+
+"""
+    SemiToBinaryBridge{T, S <: MOI.AbstractScalarSet}
+
+The `SemiToBinaryBridge` replaces an Semicontinuous constraint:
+``x \\in \\mathsf{Semicontinuous}(l, u)``
+is replaced by:
+``z \\in \\{0, 1\\}``,
+``x \\leq z \\cdot u ``,
+``x \\geq z \\cdot l ``.
+
+The `SemiToBinaryBridge` replaces an Semiinteger constraint:
+``x \\in Semiinteger(l, u)``
+is replaced by:
+``z \\in \\{0, 1\\}``,
+``x \\in \\Integer``,
+``x \\leq z \\cdot u ``,
+``x \\geq z \\cdot l ``.
+"""
+struct SemiToBinaryBridge{T, S <: SemiSets{T}} <: AbstractBridge
+    semi_set::Ref{S}
```

Comment:
Why is it using a `Ref` ?

Reply:
Otherwise I cannot modify the set, because the type is immutable.
I think it is very reasonable to allow modifying such set.

Reply:
Why not setting it mutable ?

Reply:
That would be fine for me, just didn't find any other being mutable.
I will switch.